examples/07_advanced/plot_mask_large_fmri.py

```diff
@@ -0,0 +1,412 @@
+    f"{numpy_masker['parallel']['path']} MiB"
+)
+
+
+def numpy_masker_parallel_inmemory(fmri_img, mask_imgs):
+    return Parallel(n_jobs=N_REGIONS)(
+        delayed(numpy_masker_single_inmemory)(fmri_img, mask)
+        for mask in mask_imgs
+    )
+
+
+numpy_masker["parallel"]["in_memory"] = memory_usage(
+    (numpy_masker_parallel_inmemory, (fmri_img, mask_imgs)),
+    max_usage=True,
+    include_children=True,
+    multiprocess=True,
+)
+print(
+    f"Peak memory usage: with numpy indexing, {N_REGIONS} jobs in parallel, "
+    "with in-memory images:\n"
+    f"{numpy_masker['parallel']['in_memory']} MiB"
+)
+
+# %%
+# Masking using numpy indexing in parallel with shared memory
+# -----------------------------------------------------------
+# Finally, let's see how we can mask the fMRI image using multiple masks in
+# parallel using numpy indexing and shared memory from the
+# :mod:`multiprocessing` module.
+#
+# For this method, we would have to load the fMRI image into shared memory
+# that can be accessed by multiple processes. This way, each process can
+# access the data directly from the shared memory without loading the entire
+# image into memory.
+
+from multiprocessing.shared_memory import SharedMemory
+
+fmri_array = np.asarray(fmri_img.dataobj)
+shm = SharedMemory(create=True, size=fmri_array.nbytes)
+shared_array = np.ndarray(
+    fmri_array.shape, dtype=fmri_array.dtype, buffer=shm.buf
+)
+np.copyto(shared_array, fmri_array)
+del fmri_array
+
+# %%
+# Here, the image is already in-memory, so there is no need to examine the
+# the two cases as we did before.
+
+
+def numpy_masker_shared_single(img, mask):
+    return img[np.asarray(mask.dataobj).astype(bool)]
+
+
+def numpy_masker_shared_parallel(img, masks):
+    return Parallel(n_jobs=N_REGIONS)(
+        delayed(numpy_masker_shared_single)(img, mask) for mask in masks
+    )
+
+
+numpy_masker_shared = {"single": [], "parallel": []}
+
+numpy_masker_shared["single"] = memory_usage(
+    (numpy_masker_shared_single, (shared_array, mask_imgs[0])), max_usage=True
+)
+print(
+    f"Peak memory usage: with numpy indexing, shared memory, "
+    "and single mask:\n"
+    f"{numpy_masker_shared['single']} MiB"
+)
+
+
+numpy_masker_shared["parallel"] = memory_usage(
+    (numpy_masker_shared_parallel, (shared_array, mask_imgs)),
+    max_usage=True,
+    include_children=True,
+    multiprocess=True,
+)
+print(
+    f"Peak memory usage: with numpy indexing, shared memory, "
+    f"{N_REGIONS} jobs in parallel:\n"
+    f"{numpy_masker_shared['parallel']} MiB"
+)
+
+# cleanup
+shm.close()
+shm.unlink()
+
+# %%
+# Let's plot the memory usage for each method to compare them.
+import matplotlib.pyplot as plt
+
+plt.figure(figsize=(8, 6))
+plt.bar(
+    [
+        "NiftiMasker,\nwith path",
+        "NiftiMasker,\nwith in-memory\nimage",
+        "Numpy indexing,\nwith path",
+        "Numpy indexing,\nwith in-memory\nimage",
+        "Numpy indexing\nwith shared\nmemory",
```

Comment:
To be explicit, I think the bars should be
"NiftiMasker", "ArrayProxy", "SharedMemory"